### ISSUE TITLE
Issue starting Config API running in Cloud Native

### ISSUE THREAD

**Issue by ruiengana** [JanssenProject/jans-cloud-native#17]
**Describe the bug**
Config API is not starting due to duplicate UMA scopes.

**To Reproduce**
Don't know steps to reproduce, but haven't trigger any action that could lead to duplicating scopes.

**Screenshots**
```
$ kubectl logs jans-config-api-76ccf9cbb4-w82kb -n jans
INFO - jans.pycloudlib.wait - 2021-01-18 08:39:42,957 - Config is ready
INFO - jans.pycloudlib.wait - 2021-01-18 08:39:42,986 - Secret is ready
INFO - jans.pycloudlib.wait - 2021-01-18 08:39:43,443 - LDAP is ready
08:39:48 INFO  [io.ja.co.ConfigApiApplication] (main) =================================================================
08:39:48 INFO  [io.ja.co.ConfigApiApplication] (main) =============  STARTING API APPLICATION  ========================
08:39:48 INFO  [io.ja.co.ConfigApiApplication] (main) =================================================================
08:39:48 INFO  [io.ja.co.co.ConfigurationFactory] (main) Loading base configuration /etc/jans/conf/jans.properties
08:39:48 INFO  [io.ja.co.co.ConfigurationFactory] (main) Loaded base configuration:{certsDir=/etc/certs, fido2_ConfigurationEntryDN=ou=fido2,ou=configuration,o=jans, confDir=, jansAuth_ConfigurationEntryDN=ou=jans-auth,ou=configuration,o=jans, persistence.type=ldap, pythonModulesDir=/opt/jans/python/libs:/opt/jython/Lib/site-packages}
08:39:48 INFO  [io.ja.co.co.ConfigurationFactory] (main) Loading configuration from 'ldap' DB...
08:39:48 INFO  [io.ja.or.ld.op.im.LdapConnectionProvider] (main) Attempting to create connection pool: 1
08:39:48 INFO  [io.ja.or.ld.op.im.LdapConnectionProvider] (main) Attempting to create connection pool: 1
08:39:48 INFO  [io.ja.or.ld.im.LdapEntryManagerFactory] (main) Created LdapEntryManager: io.jans.orm.ldap.operation.impl.LdapOperationServiceImpl@79aee22a
08:39:49 INFO  [io.ja.co.co.ConfigurationFactory] (main) Configuration loaded successfully.
08:39:49 INFO  [io.ja.co.co.ConfigurationFactory] (main) =============  createAuthorizationService() - ConfigurationFactory.getApiProtectionType() = oauth2
08:39:49 ERROR [io.ja.co.au.UmaResourceProtectionService] (main) 2 UMA Scope with same name.
08:39:49 ERROR [io.ja.co.co.ConfigurationFactory] (main) Failed to create AuthorizationService instance: javax.ws.rs.WebApplicationException: Multiple UMA Scope with same name - https://jans.io/oauth/config/cache.readonly
        at io.jans.configapi.auth.UmaResourceProtectionService.createScopeIfNeeded(UmaResourceProtectionService.java:104)
        at io.jans.configapi.auth.UmaResourceProtectionService.verifyResources(UmaResourceProtectionService.java:76)
        at io.jans.configapi.auth.UmaResourceProtectionService_ClientProxy.verifyResources(UmaResourceProtectionService_ClientProxy.zig:186)
        at io.jans.configapi.configuration.ConfigurationFactory.createAuthorizationService(ConfigurationFactory.java:300)
        at io.jans.configapi.configuration.ConfigurationFactory.create(ConfigurationFactory.java:158)
        at io.jans.configapi.configuration.ConfigurationFactory_ClientProxy.create(ConfigurationFactory_ClientProxy.zig:329)
        at io.jans.configapi.ConfigApiApplication.onStart(ConfigApiApplication.java:52)
        at io.jans.configapi.ConfigApiApplication_Observer_onStart_fd71b5e0b207b7d1ef838b94eaeff75e52b8f463.notify(ConfigApiApplication_Observer_onStart_fd71b5e0b207b7d1ef838b94eaeff75e52b8f463.zig:147)
        at io.quarkus.arc.impl.EventImpl$Notifier.notifyObservers(EventImpl.java:282)
        at io.quarkus.arc.impl.EventImpl$Notifier.notify(EventImpl.java:267)
        at io.quarkus.arc.impl.EventImpl.fire(EventImpl.java:69)
        at io.quarkus.arc.runtime.LifecycleEventRunner.fireStartupEvent(LifecycleEventRunner.java:23)
        at io.quarkus.arc.runtime.ArcRecorder.handleLifecycleEvents(ArcRecorder.java:60)
        at io.quarkus.deployment.steps.LifecycleEventsBuildStep$startupEvent-858218658.deploy_0(LifecycleEventsBuildStep$startupEvent-858218658.zig:81)
        at io.quarkus.deployment.steps.LifecycleEventsBuildStep$startupEvent-858218658.deploy(LifecycleEventsBuildStep$startupEvent-858218658.zig:40)
        at io.quarkus.runner.ApplicationImpl.doStart(ApplicationImpl.zig:631)
        at io.quarkus.runtime.Application.start(Application.java:90)
        at io.quarkus.runtime.ApplicationLifecycleManager.run(ApplicationLifecycleManager.java:95)
        at io.quarkus.runtime.Quarkus.run(Quarkus.java:62)
        at io.quarkus.runtime.Quarkus.run(Quarkus.java:38)
        at io.quarkus.runtime.Quarkus.run(Quarkus.java:104)
        at io.jans.configapi.MainApp.main(MainApp.java:19)

08:39:49 ERROR [io.qu.ru.Application] (main) Failed to start application (with profile prod): javax.ws.rs.WebApplicationException: Multiple UMA Scope with same name - https://jans.io/oauth/config/cache.readonly
        at io.jans.configapi.auth.UmaResourceProtectionService.createScopeIfNeeded(UmaResourceProtectionService.java:104)
        at io.jans.configapi.auth.UmaResourceProtectionService.verifyResources(UmaResourceProtectionService.java:76)
        at io.jans.configapi.auth.UmaResourceProtectionService_ClientProxy.verifyResources(UmaResourceProtectionService_ClientProxy.zig:186)
        at io.jans.configapi.configuration.ConfigurationFactory.createAuthorizationService(ConfigurationFactory.java:300)
        at io.jans.configapi.configuration.ConfigurationFactory.create(ConfigurationFactory.java:158)
        at io.jans.configapi.configuration.ConfigurationFactory_ClientProxy.create(ConfigurationFactory_ClientProxy.zig:329)
        at io.jans.configapi.ConfigApiApplication.onStart(ConfigApiApplication.java:52)
        at io.jans.configapi.ConfigApiApplication_Observer_onStart_fd71b5e0b207b7d1ef838b94eaeff75e52b8f463.notify(ConfigApiApplication_Observer_onStart_fd71b5e0b207b7d1ef838b94eaeff75e52b8f463.zig:147)
        at io.quarkus.arc.impl.EventImpl$Notifier.notifyObservers(EventImpl.java:282)
        at io.quarkus.arc.impl.EventImpl$Notifier.notify(EventImpl.java:267)
        at io.quarkus.arc.impl.EventImpl.fire(EventImpl.java:69)
        at io.quarkus.arc.runtime.LifecycleEventRunner.fireStartupEvent(LifecycleEventRunner.java:23)
        at io.quarkus.arc.runtime.ArcRecorder.handleLifecycleEvents(ArcRecorder.java:60)
        at io.quarkus.deployment.steps.LifecycleEventsBuildStep$startupEvent-858218658.deploy_0(LifecycleEventsBuildStep$startupEvent-858218658.zig:81)
        at io.quarkus.deployment.steps.LifecycleEventsBuildStep$startupEvent-858218658.deploy(LifecycleEventsBuildStep$startupEvent-858218658.zig:40)
        at io.quarkus.runner.ApplicationImpl.doStart(ApplicationImpl.zig:631)
        at io.quarkus.runtime.Application.start(Application.java:90)
        at io.quarkus.runtime.ApplicationLifecycleManager.run(ApplicationLifecycleManager.java:95)
        at io.quarkus.runtime.Quarkus.run(Quarkus.java:62)
        at io.quarkus.runtime.Quarkus.run(Quarkus.java:38)
        at io.quarkus.runtime.Quarkus.run(Quarkus.java:104)
        at io.jans.configapi.MainApp.main(MainApp.java:19)
```

**Comment by iromli** [comment]
Hi @ruiengana , can you tell us the version (tag) of config-api and persistence-loader images?

**Comment by moabu** [comment]
its _a3 @iromli .

**Comment by ruiengana** [comment]
Any update on this?
Without Config API we're locked out.

**Comment by moabu** [comment]
Yes @ruiengana . This issue should be resolved already but we haven't posted the new images as we are waiting for other updates. You may try `1.0.0_dev` image and let us know.

**Comment by ruiengana** [comment]
> Yes @ruiengana . This issue should be resolved already but we haven't posted the new images as we are waiting for other updates. You may try `1.0.0_dev` image and let us know.

Thanks @moabu! Just to be clear, it's 1.0.0_dev of config-api or persistence image?

**Comment by moabu** [comment]
It should be for all the images upon installation. So just switch out the image tags to `1.0.0_dev` until we release the a4 images. But as a warning these are updated frequently.

**Comment by ruiengana** [comment]
> It should be for all the images upon installation. So just switch out the image tags to `1.0.0_dev` until we release the a4 images. But as a warning these are updated frequently.

I am getting a error when using configuration-manager 1.0.0_dev image, from https://hub.docker.com/r/janssenproject/configuration-manager/tags?page=1&ordering=last_updated seems _dev is older than _a3. Is this correct?

**Comment by moabu** [comment]
Im not seeing the error here. Both images are on the same commit. Its just the creation date thats different .

**Comment by ruiengana** [comment]
This is one of Helm errors when moving from _a3 to _dev images.

```
Error: UPGRADE FAILED: cannot patch "jans-persistence" with kind Job: Job.batch "jans-persistence" is invalid: spec.template: Invalid value: core.PodTemplateSpec{ObjectMeta:v1.ObjectMeta{Name:"persistence", GenerateName:"", Namespace:"", SelfLink:"", UID:"", ResourceVersion:"", Generation:0, CreationTimestamp:v1.Time{Time:time.Time{wall:0x0, ext:0, loc:(*time.Location)(nil)}}, DeletionTimestamp:(*v1.Time)(nil), DeletionGracePeriodSeconds:(*int64)(nil), Labels:map[string]string{"APP_NAME":"persistence-loader", "app":"persistence", "controller-uid":"754529a5-9923-4551-bc7a-412012f1026c", "job-name":"jans-persistence"}, Annotations:map[string]string(nil), OwnerReferences:[]v1.OwnerReference(nil), Finalizers:[]string(nil), ClusterName:"", ManagedFields:[]v1.ManagedFieldsEntry(nil)}, Spec:core.PodSpec{Volumes:[]core.Volume(nil), InitContainers:[]core.Container(nil), Containers:[]core.Container{core.Container{Name:"persistence", Image:"janssenproject/persistence-loader:1.0.0_dev", Command:[]string(nil), Args:[]string(nil), WorkingDir:"", Ports:[]core.ContainerPort(nil), EnvFrom:[]core.EnvFromSource{core.EnvFromSource{Prefix:"", ConfigMapRef:(*core.ConfigMapEnvSource)(0xc004b4d480), SecretRef:(*core.SecretEnvSource)(nil)}}, Env:[]core.EnvVar(nil), Resources:core.ResourceRequirements{Limits:core.ResourceList{"cpu":resource.Quantity{i:resource.int64Amount{value:300, scale:-3}, d:resource.infDecAmount{Dec:(*inf.Dec)(nil)}, s:"300m", Format:"DecimalSI"}, "memory":resource.Quantity{i:resource.int64Amount{value:314572800, scale:0}, d:resource.infDecAmount{Dec:(*inf.Dec)(nil)}, s:"300Mi", Format:"BinarySI"}}, Requests:core.ResourceList{"cpu":resource.Quantity{i:resource.int64Amount{value:300, scale:-3}, d:resource.infDecAmount{Dec:(*inf.Dec)(nil)}, s:"300m", Format:"DecimalSI"}, "memory":resource.Quantity{i:resource.int64Amount{value:314572800, scale:0}, d:resource.infDecAmount{Dec:(*inf.Dec)(nil)}, s:"300Mi", Format:"BinarySI"}}}, VolumeMounts:[]core.VolumeMount(nil), VolumeDevices:[]core.VolumeDevice(nil), LivenessProbe:(*core.Probe)(nil), ReadinessProbe:(*core.Probe)(nil), StartupProbe:(*core.Probe)(nil), Lifecycle:(*core.Lifecycle)(nil), TerminationMessagePath:"/dev/termination-log", TerminationMessagePolicy:"File", ImagePullPolicy:"IfNotPresent", SecurityContext:(*core.SecurityContext)(0xc00a0b2a00), Stdin:false, StdinOnce:false, TTY:false}}, EphemeralContainers:[]core.EphemeralContainer(nil), RestartPolicy:"Never", TerminationGracePeriodSeconds:(*int64)(0xc02a27e8c8), ActiveDeadlineSeconds:(*int64)(nil), DNSPolicy:"ClusterFirst", NodeSelector:map[string]string(nil), ServiceAccountName:"", AutomountServiceAccountToken:(*bool)(nil), NodeName:"", SecurityContext:(*core.PodSecurityContext)(0xc019314200), ImagePullSecrets:[]core.LocalObjectReference(nil), Hostname:"", Subdomain:"", Affinity:(*core.Affinity)(nil), SchedulerName:"default-scheduler", Tolerations:[]core.Toleration(nil), HostAliases:[]core.HostAlias(nil), PriorityClassName:"", Priority:(*int32)(nil), PreemptionPolicy:(*core.PreemptionPolicy)(nil), DNSConfig:(*core.PodDNSConfig)(nil), ReadinessGates:[]core.PodReadinessGate(nil), RuntimeClassName:(*string)(nil), Overhead:core.ResourceList(nil), EnableServiceLinks:(*bool)(nil), TopologySpreadConstraints:[]core.TopologySpreadConstraint(nil)}}: field is immutable
```

**Comment by moabu** [comment]
Jobs can't get upgraded since they run once. Try deleting the persistence and config jobs and then do the upgrade.

**Comment by ruiengana** [comment]
> Jobs can't get upgraded since they run once. Try deleting the persistence and config jobs and then do the upgrade.

Deleting Jobs allows Helm to launch release, but config-api continues to fail with same error. Do I need a fresh install?

**Comment by moabu** [comment]
Yes. The operation unfortunately currently does not delete objects from the persistence layer. I think this though is a very good enhancement to have. A verification of sorts .

**Comment by ruiengana** [comment]
> Yes. The operation unfortunately currently does not delete objects from the persistence layer. I think this though is a very good enhancement to have. A verification of sorts .

Initial setup is quite hard without https://github.com/JanssenProject/jans-config-api/issues/7
and https://github.com/JanssenProject/jans-config-api/issues/6, do you know if they are included in latest _dev images?

**Comment by moabu** [comment]
Sorry @ruiengana . It hasn't been addressed yet, but there is someone working on it. As soon as its pushed we will push new images

**Comment by ruiengana** [comment]
@moabu ,  in that case I think we are going to wait for new images before cranking up a new instance.
Those two small issues really make initial setup experience a real pain.

**Comment by moabu** [comment]
Issue is resolved in 1.0.0_dev
